File: common/src/main/java/com/ultreon/mods/advanceddebug/api/common/FloatSize.java
package com.ultreon.mods.advanceddebug.api.common;

import com.ultreon.mods.advanceddebug.api.client.formatter.IFormatterContext;
import com.ultreon.mods.advanceddebug.api.util.MathUtil;

import java.util.Objects;

public final class FloatSize implements IFormattable {
    public float width;
    public float height;

    public FloatSize(float width, float height) {
        this.width = width;
        this.height = height;
    }

    public float getWidth() {
        return this.width;
    }

    public void setWidth(float width) {
        this.width = width;
    }

    public float getHeight() {
        return this.height;
    }

    public void setHeight(float height) {
        this.height = height;
    }

    @Override
    public void format(IFormatterContext ctx) {
        ctx.number(MathUtil.roundTo(this.width, 5)).operator(" × ").number(MathUtil.roundTo(this.height, 5));
    }

    @Override
    public boolean equals(Object o) {
        if (this == o) return true;
        if (o == null || this.getClass() != o.getClass()) return false;
        FloatSize size = (FloatSize) o;
        return Float.compare(size.width, this.width) == 0 && Float.compare(size.height, this.height) == 0;
    }

    @Override
    public int hashCode() {
        return Objects.hash(this.width, this.height);
    }
}

File: common/src/main/kotlin/com/ultreon/mods/advanceddebug/api/common/FloatSize.kt
package com.ultreon.mods.advanceddebug.api.common

import com.ultreon.mods.advanceddebug.api.client.formatter.IFormatterContext
import com.ultreon.mods.advanceddebug.api.util.MathUtil
import java.util.*

class FloatSize(var width: Float, var height: Float) : IFormattable {

    override fun format(ctx: IFormatterContext) {
        ctx.number(MathUtil.roundTo(width, 5)).operator(" × ").number(
            MathUtil.roundTo(
                height, 5
            )
        )
    }

    override fun equals(other: Any?): Boolean {
        if (this === other) return true
        if (other == null || javaClass != other.javaClass) return false
        val size = other as FloatSize
        return size.width.compareTo(width) == 0 && size.height.compareTo(height) == 0
    }

    override fun hashCode(): Int {
        return Objects.hash(width, height)
    }
}

val Float.squared get() = FloatSize(this, this)
fun size(width: Float, height: Float) = FloatSize(width, height)
